crates/router/src/core/payment_methods/utils.rs

```diff
@@ -0,0 +1,857 @@
+        Ok(Some(
+            builder
+                .make_all_aggregator(
+                    &[
+                        (
+                            payment_type_value_node,
+                            cgraph::Relation::Positive,
+                            cgraph::Strength::Strong,
+                        ),
+                        (
+                            connector_in_aggregator,
+                            cgraph::Relation::Positive,
+                            cgraph::Strength::Strong,
+                        ),
+                    ],
+                    None,
+                    None::<()>,
+                    None,
+                )
+                .map_err(KgraphError::GraphConstructionError)?,
+        ))
+    }
+}
+
+fn construct_card_network_nodes(
+    builder: &mut cgraph::ConstraintGraphBuilder<'_, dir::DirValue>,
+    mca_card_networks: Vec<api_enums::CardNetwork>,
+) -> Result<Option<cgraph::NodeId>, KgraphError> {
+    Ok(Some(
+        builder
+            .make_in_aggregator(
+                mca_card_networks
+                    .into_iter()
+                    .map(dir::DirValue::CardNetwork)
+                    .collect(),
+                None,
+                None::<()>,
+            )
+            .map_err(KgraphError::GraphConstructionError)?,
+    ))
+}
+
+fn compile_accepted_countries_for_mca(
+    builder: &mut cgraph::ConstraintGraphBuilder<'_, dir::DirValue>,
+    payment_method_type: &enums::PaymentMethodType,
+    pm_obj_countries: admin::AcceptedCountries,
+    config: &settings::ConnectorFilters,
+    connector: String,
+) -> Result<Option<cgraph::NodeId>, KgraphError> {
+    match pm_obj_countries {
+        admin::AcceptedCountries::EnableOnly(countries) => {
+            if let Some(config) = config
+                .0
+                .get(connector.as_str())
+                .or_else(|| config.0.get("default"))
+            {
+                if let Some(value) =
+                    config
+                        .0
+                        .get(&settings::PaymentMethodFilterKey::PaymentMethodType(
+                            *payment_method_type,
+                        ))
+                {
+                    // Country from the MCA
+                    let pm_object_country_value_node = builder
+                        .make_in_aggregator(
+                            countries
+                                .into_iter()
+                                .map(|country| {
+                                    dir::DirValue::BillingCountry(
+                                        common_enums::Country::from_alpha2(country),
+                                    )
+                                })
+                                .collect(),
+                            None,
+                            None::<()>,
+                        )
+                        .map_err(KgraphError::GraphConstructionError)?;
```

Comment:
This node should be created regardless of whether the `connector.payment_method_type` config exists or not right? So this should probably happen outside the `if let`